depinject/appconfig/dynamic_resolver.go

```diff
@@ -0,0 +1,80 @@
+package appconfig
+
+import (
+	"strings"
+
+	"google.golang.org/protobuf/reflect/protodesc"
+	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/reflect/protoregistry"
+	"google.golang.org/protobuf/types/dynamicpb"
+)
+
+// dynamic resolver allows marshaling gogo proto messages from the gogoproto.HybridResolver as long as those
+// files have been imported before calling LoadJSON. There is similar code in autocli, this should probably
+// eventually be moved into a library.
```

Comment:
Do we want to refactor this code and the almost identical implementation in autocli to cosmos-proto possibly?

Reply:
We could also move it to gogoproto because it uses the hybrid resolver and then call it here: https://github.com/cosmos/gogoproto/blob/6eec9731781bf8ec4feb0e4098ffb19ee74afb50/jsonpb/jsonpb.go#L792. So that gogoproto jsonpb.Unmarshal can be used to unmarshal v2 messages from json that contain some gogo proto messages in Any's.